go/state/state_db_test.go

```diff
@@ -4143,6 +4143,70 @@ func TestSlotIdOrder(t *testing.T) {
 	}
 }
 
+func TestStateDB_SetTransientState_NewValue_UndoRemovesValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := createStateDBWith(mock, defaultStoredDataCacheSize, true)
+
+	db.SetTransientState(address1, key1, val1)
+
+	// Call undo
+	for _, f := range db.undo {
+		f()
+	}
+
+	got := db.GetTransientState(address1, key1)
```

Comment:
Please use following syntax to test got/wanted value in tests - here and in other tests 
```

if got, want := db.GetTransientState(address1, key1), val2; got != want {
		t.Errorf("unexpected value, wanted %v, got %v", want, got)
	}
```

notice:
* got, want is assigned and checked inline
* error message starts with lower case and there is no perioed at the end of the sentece